crates/milli/src/search/new/ranking_rule_graph/fid/mod.rs

```diff
@@ -88,10 +88,10 @@ impl RankingRuleGraphTrait for FidGraph {
         }
 
         // always lookup the max_fid if we don't already and add an artificial condition for max scoring
-        let max_weight: Option<u16> = weights_map.max_weight();
+        let max_weight_fid = weights_map.max_weight_fid();
 
-        if let Some(max_weight) = max_weight {
-            if !all_fields.contains(&max_weight) {
+        if let Some((max_fid, max_weight)) = max_weight_fid {
+            if !all_fields.contains(&max_fid) {
```

Comment:
Thank you for fixing this!

Note that we probably want the condition to check if any fid with the max_weight exists in the edges, rather than just checking for `max_fid`, so that we only add the condition when **no** other fid has the `max_weight`, but that is already fixing a bug.

Reply:
true, I'll try to enhance that